lib/src/test/java/io/ably/lib/test/realtime/RealtimeConnectFailTest.java

```diff
@@ -166,23 +167,32 @@ public void connect_while_disconnected() {
 	}
 
 	/**
-	 * Verify that the connection enters the disconnected state, after a token
-	 * used for successful connection expires
+	 * Verify that the server issues reauth message 30 seconds before token expiration time, authCallback is
+	 * called to obtain new token and in-place re-authorization takes place with connection staying in connected
+	 * state
 	 */
 	@Test
-	public void connect_token_expire_disconnected() {
+	public void connect_token_expire_inplace_reauth() {
```

Comment:
Whilst I see the point of changing this test, we do need to know still that a client does not authenticate in response to the server does move to the disconnected state and then attempt a reconnect.  I would prefer to have two tests in fact to make sure we're not breaking the behaviour whereby a token expires without the change to renew (or perhaps with a failed renew etc).